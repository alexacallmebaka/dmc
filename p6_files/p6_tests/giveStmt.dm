a:int;
b:int = 1;
c:bool = true;
main : () void {
  give "hello";
  give b;
}