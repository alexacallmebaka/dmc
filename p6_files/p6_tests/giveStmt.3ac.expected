[BEGIN GLOBALS]
a
b
c
main
str_0 "hello"
[END GLOBALS]
[BEGIN <init> LOCALS]
[END <init> LOCALS]
fun_<init>: enter <init>
            [b] := 1
            [c] := 1
lbl_0:      leave <init>
[BEGIN main LOCALS]
[END main LOCALS]
main:       enter main
            WRITE str_0
            WRITE [b]
lbl_1:      leave main